### PR TITLE
perf: Avoid pooling for long-living `HashtableEx`es

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

### <samp>🤖 Generated by Copilot at 29883f5</samp>

This pull request introduces a new benchmark for hash table implementations and changes the `PropertyMetadataDictionary` class to use the standard `Hashtable` instead of the custom `HashtableEx`. The goal is to evaluate the performance and memory impact of different hash table options in the Uno platform.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/DependencyProperty.Dictionary.FrameworkPropertiesForType.cs [hunk 1]
```diff
@@ -13,18 +13,19 @@
 using Uno.Collections;
 
 using _Key = Uno.CachedTuple<System.Type, Windows.UI.Xaml.FrameworkPropertyMetadataOptions>;
+using System.Collections;
 
 namespace Windows.UI.Xaml
 {
 	public sealed partial class DependencyProperty
 	{
 		private class FrameworkPropertiesForTypeDictionary
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/DependencyProperty.Dictionary.FrameworkPropertiesForType.cs:22]: This dictionary has a single static instance that is kept for the lifetime of the whole app.

File: src/Uno.UI/UI/Xaml/DependencyProperty.Dictionary.NameToProperty.cs [hunk 2]
```diff
@@ -26,11 +27,11 @@ public sealed partial class DependencyProperty
 	{
 		private class NameToPropertyDictionary
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/DependencyProperty.Dictionary.NameToProperty.cs:28]: This dictionary has a single static instance that is kept for the lifetime of the whole app.

File: src/Uno.UI/UI/Xaml/DependencyProperty.Dictionary.PropertiesRegistry.cs [hunk 3]
```diff
@@ -12,14 +12,15 @@
 using System.Threading;
 using Uno.Collections;
 using Uno.UI.Helpers;
+using System.Collections;
 
 namespace Windows.UI.Xaml
 {
 	public sealed partial class DependencyProperty
 	{
 		private class DependencyPropertyRegistry
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/DependencyProperty.Dictionary.PropertiesRegistry.cs:21]: This dictionary has a single static instance that is kept for the lifetime of the whole app.

File: src/Uno.UI/UI/Xaml/DependencyProperty.Dictionary.TypeNullableDictionary.cs [hunk 4]
```diff
@@ -23,11 +24,11 @@ private bool GetIsTypeNullable(Type type)
 
 		private class TypeNullableDictionary
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/DependencyProperty.Dictionary.TypeNullableDictionary.cs:25]: This dictionary has a single static instance that is kept for the lifetime of the whole app.

File: src/Uno.UI/UI/Xaml/DependencyProperty.Dictionary.TypeToProperties.cs [hunk 5]
```diff
@@ -10,11 +11,11 @@ public sealed partial class DependencyProperty
 	{
 		private class TypeToPropertiesDictionary
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/DependencyProperty.Dictionary.TypeToProperties.cs:12]: This dictionary has a single static instance that is kept for the lifetime of the whole app.

File: src/Uno.UI/UI/Xaml/PropertyMetadataDictionary.cs [hunk 6]
```diff
@@ -20,17 +20,16 @@ internal class PropertyMetadataDictionary
 		/// This implementation of PropertyMetadataDictionary uses HashTable because it's generally faster than
 		/// Dictionary`ref/ref. 
 		/// </summary>
-		private readonly HashtableEx _table = new HashtableEx();
+		private readonly Hashtable _table = new Hashtable();
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/PropertyMetadataDictionary.cs:23]: This is created per dependency property instance. So there is a hashtable for each DP, and it's likely that all these instances live for the lifetime of the app.

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**